manifests/params.pp

```diff
@@ -106,6 +106,7 @@
         'php5-ldap',
         'php5-mysql',
         'php5-pgsql',
+        'mysql-client-core-5.5',
```

Comment:
Nobody forces you to use the server part of the module.
I can't speak for every module but at least with aforementioned one you can ```include mysql::client``` to have only the client binaries installed.